pkg/auth/signing/encode.go

```diff
@@ -88,3 +88,13 @@ func EncodeVoucher(ctx context.Context, voucher *pb.Voucher) (_ []byte, err erro
 	voucher.SatelliteSignature = signature
 	return out, err
 }
+
+// EncodeStreamID encodes stream ID into bytes for signing.
+func EncodeStreamID(ctx context.Context, streamID *pb.SatStreamID) (_ []byte, err error) {
+	defer mon.Task()(&ctx)(&err)
+	signature := streamID.SatelliteSignature
+	streamID.SatelliteSignature = nil
```

Comment:
I wonder if this could cause race issues. What if another goroutine is trying to access the SatelliteSignature while we are marshaling the `streamID`? Wouldn't it be safer if we copy `streamID` to a new struct and then marshal it?

Reply:
not sure tbh, I guess will have a problem always when we will try to use `streamID` in different gouroutines.